### DISCUSSION TITLE
Reference a .NET Framework library

### DISCUSSION THREAD

**Discussion by chhayes** [microsoft/WindowsAppSDK#1857] [Q&A] [answered]
A project I'm working on has a non-negotiable dependency on a .NET Framework library (sadly, I have no access to the source code). Currently, I'm using .NET Core 3.1/WPF, and I was able to get the library to work using the <UseWindowsForms> flag in the project file.

Unfortunately, I've had less luck with a Windows App SDK project. If I set the <UseWindowsForms> flag to true, I get the following compiler error.

`C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk.WindowsDesktop\targets\Microsoft.WinFX.targets(223,9): error MC6000: Project file must include the .NET Framework assembly 'PresentationCore, PresentationFramework' in the reference list.`

I tried setting the <UseWPF> flag as well to resolve this. While this does correct the above compiler error, the compiler then begins to throw XAML syntax errors. Manually adding a reference to PresentationCore.dll and PresentationFramework.dll does not seem to work either.

Is there any way I can get this library functional in a WinUI 3 app? Or will I need to stick with WPF for the time being?

Thanks in advance!

**Comment by andrewleader** [comment, marked as answer]
Hi @chhayes, it sounds like the library you're using uses WPF (PresentationCore and PresentationFramework). Using WPF from within WinUI 3 apps currently isn't supported. It's something that might be supported someday through what's called "reverse XAML Islands" (allowing you to use WPF from within WinUI 3), but we don't have any immediate plans to implement that.

**Reply by chhayes** [reply]
Hi @andrewleader, I don't believe the library has any WPF dependencies, it was originally designed for WinForms applications (I suspect it only uses the WinForms MessageBox, but I am unable to confirm).

That being said, I get the PresentationFramework/PresentationCore error before referencing the library at all. If you create a WinUI 3 app using the "Blank App, Packaged (WinUI3 in Desktop)" template in VS2022, and add `<UseWindowsForms>true</UseWindowsForms>` to the project file, the PresentationFramework compiler error is thrown.

This doesn't seem to be the case in other application types. Adding the WinForms flag to a .NET 6 console application does not cause the same compiler error. And indeed, with only the WinForms flag set, the library works as expected from the console application.

**Reply by andrewleader** [reply]
Using either Windows Forms (`<UseWindowsForms>true</UseWindowsForms>`) or WPF with a WinUI 3 app is not supported.

I'll check with the team whether features that only use something like WinForms MessageBox still requires full "reverse XAML Islands", but either way supporting using Windows Forms or WPF APIs from within a WinUI 3 app would unfortunately be a new feature that currently isn't in our 1.1 plans.

**Reply by chhayes** [reply]
Understood, thank you for the clarification.
